packages/sourcecred/src/plugins/discord/createGraph.js

```diff
@@ -320,48 +320,53 @@ export function _createGraphFromMessages(
       channelId,
       channelParentId,
     } = graphMessage;
-    if (author) {
-      wg.graph.addNode(memberNode(author));
-      wg.graph.addEdge(authorsMessageEdge(message, author));
-    }
-    wg.graph.addNode(messageNode(message, guildId, channelName));
 
+    let messageWeight = 0;
     for (const {reaction, reactingMember} of reactions) {
-      const node = reactionNode(reaction, message.timestampMs, guildId);
-      wg.weights.nodeWeights.set(
-        node.address,
-        reactionWeight(
-          weights,
-          message,
-          reaction,
-          reactingMember,
-          propsChannels,
-          reactions,
-          channelParentId
-        )
-      );
-      wg.graph.addNode(node);
-      wg.graph.addNode(memberNode(reactingMember));
-      wg.graph.addEdge(reactsToEdge(reaction, message));
-      wg.graph.addEdge(
-        addsReactionEdge(reaction, reactingMember, message.timestampMs)
-      );
+      const weight = reactionWeight({
+        weights,
+        message,
+        reaction,
+        reactingMember,
+        propsChannels,
+        reactions,
+        channelParentId,
+      });
+      messageWeight += weight;
+      if (weight && !config.simplifyGraph) {
```

Comment:
OK I think I'm getting the hang of this change now. 

I would probably put the simple case first here: if (simplifyGraph) { ... } else { ...complicated stuff ... }

I might even use `if (!messageWeight) continue` here near the top of the loop, because it would seem more obvious.  I don't know if you'd treat `continue` like a smell, though.  WDYT?

Reply:
I like it